Code/GraphMol/SubstructLibrary/SubstructLibrary.h

```diff
@@ -286,6 +295,20 @@ class RDKIT_SUBSTRUCTLIBRARY_EXPORT PatternHolder : public FPHolderBase {
   }
 };
 
+//! Uses the pattern fingerprinter with a user-defined number of bits to rule out matches
+class RDKIT_SUBSTRUCTLIBRARY_EXPORT PatternNumBitsHolder : public FPHolderBase {
```

Comment:
I'd rather not make two classes, let me see if I can get it to work with only one.

Reply:
@bp-kelley That’s doable, but I think you need to bump the version number if you wish to keep backwards compatibility when deserialising older `PatternHolder`s (I wasn’t sure how you felt about that). Or (less elegant) catch the exception when trying to read `numBits` past the archive end when deserialising older objects.